### DISCUSSION TITLE
Call custom capacitor plugin from another class

### DISCUSSION THREAD

**Discussion by mwilogis** [ionic-team/capacitor#5493] [Q&A] [answered]
In capacitor 3, I've build a minimal plugin in Android which can send events from the native Android to javascript, whenever something is happening, by listening for events in javascript.

It's working fine from within the class, i.e. overriding OnStart of the class. But when I want to call the class and notify listeners from another class, how do I call the plugin class without making a new instance, so that whoever is listening, can still get the event.

I tried implementing it as singleton, but it is not allowed, I get a error in javascript error handler when doing so, just when I start listening to events. Also I can't make it static.

So how do I call the plugin class without making a new instance?

The javascriptcode:

```
IonicNativePlugin.addListener('EVENT_LISTENER_NAME', (message) => {
      alert(message.message);
    }).catch((error) => {
      alert('Error in EventListener');
    });
```
    
The java MainActivity code:

```
public class MainActivity extends BridgeActivity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
            registerPlugin(IonicNativeLogisPlugin.class);
    }
}
```

The plugin java code:

```
@CapacitorPlugin(name = "IonicNativePlugin")
public class IonicNativePlugin extends Plugin {
  @PluginMethod
  public void NotifyListeners(PluginCall call){
    JSObject result = new JSObject();
    result.put("message", "Hello!");
    notifyListeners("EVENT_LISTENER_NAME", result);
  }

  public void TheMethodIWantToCall(){
    JSObject ret = new JSObject();
    ret.put("message", "Hello from LISTENER!!");
    notifyListeners("EVENT_LISTENER_NAME", ret);
  }
}
```

The class I like to call from:

```
public class NotificationServiceExtension implements OSRemoteNotificationReceivedHandler {

  @Override
  public void remoteNotificationReceived(Context context, OSNotificationReceivedEvent notificationReceivedEvent) {
    OSNotification notification = notificationReceivedEvent.getNotification();
    notificationReceivedEvent.complete(notification);
    <-- call should be here
  }
}
```

**Comment by jcesarmobile** [comment, marked as answer]
You can use a static method like the push plugin does
https://github.com/ionic-team/capacitor-plugins/blob/main/push-notifications/android/src/main/java/com/capacitorjs/plugins/pushnotifications/MessagingService.java#L18 
https://github.com/ionic-team/capacitor-plugins/blob/22fb04179bd13fabea6187c42ed3ffc2faa866c1/push-notifications/android/src/main/java/com/capacitorjs/plugins/pushnotifications/PushNotificationsPlugin.java#L203

**Reply by mwilogis** [reply]
@jcesarmobile 

In "PushNotificationsPlugin.java" there is a "load()" method which sets up the "staticBridge". But I can't see where the load method is called.

Thanks for the help buddy.

**Reply by jcesarmobile** [reply]
it's called automatically on Capacitor initialization 
https://github.com/ionic-team/capacitor/blob/b893a57aaaf3a16e13db9c33037a12f1a5ac92e0/android/capacitor/src/main/java/com/getcapacitor/PluginHandle.java#L95

**Reply by mwilogis** [reply]
It runs perfectly. Thanks for clarifying.

**Reply by mwilogis** [reply]
@jcesarmobile Hi can I get you to show me the implementation for ios? Don't seem to find the code I need inside of the ios folder.

**Reply by mwilogis** [reply]
@jcesarmobile I have created a new discussion regarding the iOS part, to be more specific.

#5512
